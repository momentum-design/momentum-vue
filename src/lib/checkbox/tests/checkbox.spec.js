import { mount } from '@vue/test-utils';
import Checkbox from '../index.vue';
import Label from '../../label/index.vue';
import Vue from 'vue';

Vue.component(Label.name, Label);

describe('Checkbox', () => {
  it('should match snapshot', () => {
    const wrapper = mount(Checkbox, {
      propsData: {
        htmlId: 'test123',
        label: 'test'
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render one Checkbox', () => {
    const wrapper = mount(Checkbox, {
      propsData: {
        htmlId: 'test123',
        label: 'test'
      }
    });

    expect(wrapper.findAll('input').length).toEqual(1);
    expect(wrapper.findAll('label').length).toEqual(1);
  });

  it('should render Label', () => {
    const wrapper = mount(Checkbox, {
      propsData: {
        htmlId: 'test123',
        label: 'test'
      }
    });

    expect(wrapper.contains(Label)).toEqual(true);
  });

  it('should pass class based on indeterminate prop', () => {
    const wrapper = mount(Checkbox, {
      propsData: {
        htmlId: 'test123',
        label: 'test',
        indeterminate: true
      }
    });

    expect(wrapper.find('input').classes('indeterminate')).toEqual(true);
  });

  it('should pass class based on nesting', () => {
    const wrapper = mount(Checkbox, {
      propsData: {
        htmlId: 'test123',
        label: 'test',
        nestedLevel: 1
      }
    });

    expect(wrapper.classes('md-input--nested-1')).toEqual(true);
  });

  it('should handle disabled state', () => {
    const wrapper = mount(Checkbox, {
      propsData: {
        htmlId: 'test123',
        label: 'test',
        disabled: true
      }
    });

    expect(wrapper.find('input').attributes().disabled).toEqual('disabled');
  });

  it('should render children', () => {
    const wrapper = mount(Checkbox, {
      propsData: {
        htmlId: 'test123',
        label: 'test',
      },
      slots: {
        default: '<div class="test-text">I\'m here to help</div>'
      }
    });

    expect(wrapper.findAll('.test-text').length).toEqual(1);
    expect(wrapper.find('.test-text').text()).toEqual("I'm here to help");
  });

  it('should handle focus events if passed', () => {
    let count = 0;
    const countUp = () => count++;
    const wrapper = mount(Checkbox, {
      propsData: {
        htmlId: 'test123',
        label: 'test',
      },
      listeners: {
        focus: countUp
      }
    });

    wrapper.find('input').trigger('focus');
    expect(count).toEqual(1);
  });

  it('should handle change event', () => {
    let count = 0;
    const countUp = () => count++;
    const wrapper = mount(Checkbox, {
      propsData: {
        htmlId: 'test123',
        label: 'test',
      },
      listeners: {
        change: countUp
      }
    });

    wrapper.find('input').trigger('change');
    expect(count).toEqual(1);
  });

  it('should support input ref', () => {
    const inputRef = 'input1';
    const wrapper = mount(Checkbox, {
      propsData: {
        htmlId: 'test123',
        label: 'test',
        inputRef: inputRef
      }
    });

    expect(wrapper.vm.$refs[inputRef].tagName).toEqual('INPUT');
  });

});

