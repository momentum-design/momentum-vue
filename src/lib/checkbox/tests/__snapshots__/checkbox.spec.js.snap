// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should match snapshot 1`] = `<div class="md-input-group md-checkbox"><input type="checkbox" id="test123" name="" tabindex="0" class="md-input md-checkbox__input" value=""> <label for="test123" class="md-checkbox__label"><span>test</span></label> </div>`;
